features/continue_as_new/continue_as_same/feature.go

```diff
@@ -0,0 +1,72 @@
+package continue_as_same
+
+import (
+	"context"
+	"time"
+
+	"go.temporal.io/sdk-features/harness/go/harness"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/converter"
+	"go.temporal.io/sdk/workflow"
+)
+
+const (
+	InputData  = "InputData"
+	MemoKey    = "MemoKey"
+	MemoValue  = "MemoValue"
+	WorkflowID = "TestID"
+)
+
+var Feature = harness.Feature{
+	Workflows: Workflow,
+	CheckResult: func(ctx context.Context, r *harness.Runner, run client.WorkflowRun) error {
+		var result string
+		if err := run.Get(ctx, &result); err != nil {
+			return err
+		}
+		r.Require.Equal(InputData, result)
+
+		execution, err := r.Client.DescribeWorkflowExecution(ctx, run.GetID(), run.GetRunID())
+		if err != nil {
+			return err
+		}
+		// Workflow ID does not change after continue as new
+		r.Require.Equal(WorkflowID, run.GetID())
+		// Memos do not change after continue as new
+		memo := execution.GetWorkflowExecutionInfo().GetMemo().GetFields()
+		r.Require.NotNil(memo)
+		memoPayload, ok := memo[MemoKey]
+		r.Require.True(ok)
+		var testMemo string
+		err = converter.GetDefaultDataConverter().FromPayload(memoPayload, &testMemo)
+		if err != nil {
+			return err
+		}
+		r.Require.Equal(MemoValue, testMemo)
+		return nil
```

Comment:
All of this isn't necessary IMO, it's enough to check the memo from the workflow